pkg/cache/v3/simple.go

```diff
@@ -306,6 +306,34 @@ func (cache *snapshotCache) CreateWatch(request *Request, value chan Response) f
 	snapshot, exists := cache.snapshots[nodeID]
 	version := snapshot.GetVersion(request.TypeUrl)
 
+	if exists {
+		knownResourceNames := streamState.GetKnownResourceNames(request.TypeUrl)
+		diff := []string{}
+		for _, r := range request.ResourceNames {
+			if _, ok := knownResourceNames[r]; !ok {
+				diff = append(diff, r)
+			}
+		}
+		if cache.log != nil {
```

Comment:
Sorry you can remove this nil check here. We no longer require that

Reply:
done